python/ray/_private/runtime_env/conda_utils.py

```diff
@@ -276,3 +276,13 @@ def exec_cmd_stream_to_logger(
 
     exit_code = child.wait()
     return exit_code, "\n".join(last_n_lines)
+
+
+def check_if_conda_environment_valid(conda_env_name: str) -> bool:
+    """
+    Check if the given conda environment exists and is valid.
+    We try to `conda activate` it and check that python is runnable
+    """
+    activate_cmd = get_conda_activate_commands(conda_env_name) + ["python", "--version"]
```

Comment:
Do we need to deactivate it afterwards?

Reply:
I don't believe so, because conda activate sets environment variables and other things that aren't global state, which will be reset when the subprocess exits.

However, I did recently do some thinking about it, and I think conda activate does a lot of shell-specific things, so I've changed it to execute in a shell.